### PR TITLE
use LazyUser in DefaultShareProvider

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1:

File: lib/private/Share20/DefaultShareProvider.php [hunk 1]
```diff
@@ -946,8 +947,8 @@ public function getSharedWith($userId, $shareType, $node, $limit, $offset) {
 			}
 			$cursor->closeCursor();
 		} elseif ($shareType === IShare::TYPE_GROUP) {
-			$user = $this->userManager->get($userId);
-			$allGroups = ($user instanceof IUser) ? $this->groupManager->getUserGroupIds($user) : [];
+			$user = new LazyUser($userId, $this->userManager);
```
[the remark this answers is not in the document]
Comment (icewind1991) [reply lib/private/Share20/DefaultShareProvider.php:950]: `LazyUser` doesn't ask user backends for user metadata until requested, and uses the `DisplayManagerCache`.

This prevents having to load user info from the backends in most cases

**Review by CarlSchwan — APPROVED** [review]

**Comment by nickvergessen** [comment]
> Tests: 4297, Assertions: 15882, Failures: 11, Skipped: 28, Risky: 1.

DB unit tests are failing

**Comment by PVince81** [comment]
@icewind1991 please fix the tests

**Comment by come-nc** [comment]
Rebased and fixed conflicts

**Review by come-nc — APPROVED** [review]

**Pull request merged**